diff -up tachyon/unix/Make-arch.r tachyon/unix/Make-arch
--- tachyon/unix/Make-arch.r	2013-04-21 21:09:28.000000000 +0200
+++ tachyon/unix/Make-arch	2013-08-09 00:26:25.175778876 +0200
@@ -11,7 +11,7 @@ RANLIB= touch
 
 MISCDEFS=$(USEJPEG) $(USEPNG) $(FLT) $(MBOX)
 MISCINC=$(JPEGINC) $(PNGINC) $(SPACEBALLINC)
-MISCFLAGS=$(MISCDEFS) $(MISCINC)
+MISCFLAGS=$(MISCDEFS) $(MISCINC) $(OPTFLAGS)
 MISCLIB=$(JPEGLIB) $(PNGLIB) $(SPACEBALLLIB)
 
 default:
@@ -1217,10 +1217,10 @@ linux-64-thr:
 	$(MAKE) all \
 	"ARCH = linux-64-thr" \
 	"CC = gcc" \
-	"CFLAGS = -m64 -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux  $(THREADSFLAGS) -D_REENTRANT $(MISCFLAGS)" \
+	"CFLAGS = -m64 -Wall -O2 -DLinux $(THREADSFLAGS) -D_REENTRANT $(MISCFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
-	"STRIP = strip" \
+	"STRIP = touch" \
 	"RANLIB = ranlib" \
 	"LIBS = -L. -ltachyon $(MISCLIB) -lm -lpthread"
 
@@ -1364,10 +1364,10 @@ linux-thr:
 	$(MAKE) all \
 	"ARCH = linux-thr" \
 	"CC = gcc" \
-	"CFLAGS = -m32 -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux $(THREADSFLAGS) -D_REENTRANT $(MISCFLAGS)" \
+	"CFLAGS = -m32 -Wall -O2 -DLinux $(THREADSFLAGS) -D_REENTRANT $(MISCFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
-	"STRIP = strip" \
+	"STRIP = touch" \
 	"RANLIB = ranlib" \
 	"LIBS = -L. -ltachyon $(MISCLIB) -lm -lpthread"
 
@@ -1376,7 +1376,7 @@ linux-thr-ogl:
 	$(MAKE) all \
 	"ARCH = linux-thr-ogl" \
 	"CC = cc" \
-	"CFLAGS = -m32 -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux $(THREADSFLAGS) -D_REENTRANT $(MISCFLAGS) -DUSEOPENGL $(LINUX_GLX_INCS)" \
+	"CFLAGS = -m32 -Wall -O2 -DLinux $(THREADSFLAGS) -D_REENTRANT $(MISCFLAGS) -DUSEOPENGL $(LINUX_GLX_INCS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = touch" \
@@ -1387,7 +1387,7 @@ linux-64-thr-ogl:
 	$(MAKE) all \
 	"ARCH = linux-64-thr-ogl" \
 	"CC = cc" \
-	"CFLAGS = -m64 -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux $(THREADSFLAGS) -D_REENTRANT $(MISCFLAGS) -DUSEOPENGL $(LINUX_GLX_INCS)" \
+	"CFLAGS = -m64 -Wall -O2 -DLinux $(THREADSFLAGS) -D_REENTRANT $(MISCFLAGS) -DUSEOPENGL $(LINUX_GLX_INCS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = touch" \
@@ -1446,25 +1446,37 @@ linux-beowulf-mpi:
 linux-lam:
 	$(MAKE) all \
 	"ARCH = linux-lam" \
-	"CC = hcc" \
-	"CFLAGS = -m486 -Wall -O3 -fomit-frame-pointer -ffast-math -I$(LAMHOME)/h -DLinux $(MISCFLAGS) $(MPIFLAGS)" \
+	"CC = mpicc" \
+	"CFLAGS = -m32 -Wall -O2 $(LAMFLAGS) -DLinux $(MISCFLAGS) $(MPIFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
-	"STRIP = strip" \
+	"STRIP = touch" \
+	"RANLIB = ranlib" \
+	"LIBS = -L. -ltachyon $(LAMLIBS) $(MISCLIB) -lm"
+
+# Linux x86 using LAM MPI, with OpenGL display
+linux-lam-ogl:
+	$(MAKE) all \
+	"ARCH = linux-lam-ogl" \
+	"CC = mpicc" \
+	"CFLAGS = -m32 -Wall -O2 $(LAMFLAGS) -DLinux $(MISCFLAGS) $(MPIFLAGS) -DUSEOPENGL $(LINUX_GLX_INCS)" \
+	"AR = ar" \
+	"ARFLAGS = r" \
+	"STRIP = touch" \
 	"RANLIB = ranlib" \
-	"LIBS = -L. -L$(LAMHOME)/lib -ltachyon -lmpi $(MISCLIB) -lm"
+	"LIBS = -L. -ltachyon $(LAMLIBS) $(LINUX_GLX_LIBS) $(MISCLIB) -lm"
 
 # Linux AMD64/EM64T using LAM MPI
-linux-lam-64:
+linux-64-lam:
 	$(MAKE) all \
-	"ARCH = linux-lam-64" \
-	"CC = hcc" \
-	"CFLAGS = -m64 -Wall -O3 -fomit-frame-pointer -ffast-math -I$(LAMHOME)/h -DLinux $(MISCFLAGS) $(MPIFLAGS)" \
+	"ARCH = linux-64-lam" \
+	"CC = mpicc" \
+	"CFLAGS = -m64 -Wall -O2 $(LAMFLAGS) -DLinux -DLP64 $(MISCFLAGS) $(MPIFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
-	"STRIP = strip" \
+	"STRIP = touch" \
 	"RANLIB = ranlib" \
-	"LIBS = -L. -L$(LAMHOME)/lib -ltachyon -lmpi $(MISCLIB) -lm"
+	"LIBS = -L. -ltachyon $(LAMLIBS) $(MISCLIB) -lm"
 
 linux-mpi:
 	$(MAKE) all \
@@ -1524,16 +1536,16 @@ linux-mpi-ogl:
 	"LIBS = -L. -L$(LAMHOME)/lib -ltachyon -lmpi $(LINUX_GLX_LIBS) $(MISCLIB) -lm"
 
 # Linux AMD64/EM64T using LAM MPI, with OpenGL display
-linux-lam-64-ogl:
+linux-64-lam-ogl:
 	$(MAKE) all \
-	"ARCH = linux-lam-64-ogl" \
-	"CC = hcc" \
-	"CFLAGS = -m64 -Wall -O3 -fomit-frame-pointer -ffast-math -I$(LAMHOME)/h -DLinux -DUSEOPENGL $(LINUX_GLX_INCS) $(MISCFLAGS) $(MPIFLAGS)" \
+	"ARCH = linux-64-lam-ogl" \
+	"CC = mpicc" \
+	"CFLAGS = -m64 -Wall -O2 $(LAMFLAGS) -DLinux -DUSEOPENGL $(LINUX_GLX_INCS) $(MISCFLAGS) $(MPIFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
-	"STRIP = strip" \
+	"STRIP = touch" \
 	"RANLIB = ranlib" \
-	"LIBS = -L. -L$(LAMHOME)/lib -ltachyon -lmpi $(LINUX_GLX_LIBS) $(MISCLIB) -lm"
+	"LIBS = -L. -ltachyon $(LAMLIBS) $(LINUX_GLX_LIBS) $(MISCLIB) -lm"
 
 # Linux x86 using LAM MPI and threads
 linux-lam-thr:
diff -up tachyon/unix/Make-config.r tachyon/unix/Make-config
--- tachyon/unix/Make-config.r	2013-04-21 21:09:28.000000000 +0200
+++ tachyon/unix/Make-config	2013-08-09 00:20:28.804492178 +0200
@@ -105,7 +105,7 @@ THREADSFLAGS=-DTHR -DUSEATOMICBARRIERS
 ##########################################################################
 # Customize MPI directories and includes as-needed.
 # A typical MPICH installation location:
-MPIDIR=/usr/local/mpi
+MPIDIR=/usr
 MPIINC=$(MPIDIR)/include
 MPILIB=$(MPIDIR)/lib
 
@@ -125,7 +125,9 @@ MPIFLAGS=-DMPI -DUSE_MPI_IN_PLACE
 # MPI 1.x flags:
 #MPIFLAGS=-DMPI 
 
-
+# LAM
+LAMFLAGS=`pkg-config --cflags lam`
+LAMLIBS=`pkg-config --libs lam`
 
 ##########################################################################
 # Floating point configuration:
@@ -166,14 +168,14 @@ MBOX=
 #   http://www.ijg.org/files/
 ##########################################################################
 # Uncomment the following lines to disable JPEG support
-USEJPEG=
-JPEGINC=
-JPEGLIB=
+#USEJPEG=
+#JPEGINC=
+#JPEGLIB=
 
 # Uncomment the following lines to enable JPEG support
-#USEJPEG= -DUSEJPEG
-#JPEGINC= -I/usr/local/include
-#JPEGLIB= -L/usr/local/lib -ljpeg
+USEJPEG= -DUSEJPEG
+JPEGINC= 
+JPEGLIB= -ljpeg
 
 
 ##########################################################################
@@ -186,14 +188,14 @@ JPEGLIB=
 #   http://www.libpng.org/
 ##########################################################################
 # Uncomment the following lines to disable PNG support
-USEPNG=
-PNGINC=
-PNGLIB=
+#USEPNG=
+#PNGINC=
+#PNGLIB=
 
 # Uncomment the following lines to enable PNG support
-#USEPNG= -DUSEPNG
-#PNGINC= -I/usr/local/include
-#PNGLIB= -L/usr/local/lib -lpng -lz
+USEPNG= -DUSEPNG
+PNGINC= `pkg-config --cflags libpng`
+PNGLIB= `pkg-config --libs libpng`
 
 
 ##########################################################################
