diff -p -up tachyon/unix/Make-arch.orig tachyon/unix/Make-arch
--- tachyon/unix/Make-arch.orig	2010-01-18 17:13:41.000000000 -0200
+++ tachyon/unix/Make-arch	2010-11-22 20:25:50.022869977 -0200
@@ -1069,7 +1069,7 @@ linux-64-thr:
 	$(MAKE) all \
 	"ARCH = linux-64-thr" \
 	"CC = gcc" \
-	"CFLAGS = -m64 -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux -DLP64 -DTHR -D_REENTRANT $(MISCFLAGS)" \
+	"CFLAGS = -fPIC -m64 -Wall -O3 -fomit-frame-pointer -ffast-math -DLinux -DLP64 -DTHR -D_REENTRANT $(MISCFLAGS)" \
 	"AR = ar" \
 	"ARFLAGS = r" \
 	"STRIP = strip" \
